corehq/apps/hqwebapp/static/hqwebapp/js/urllib.js

```diff
@@ -0,0 +1,37 @@
+var COMMCAREHQ_URLS = {};
+hqDefine('hqwebapp/js/urllib.js', function () {
+    // http://stackoverflow.com/a/21903119/240553
+    var getUrlParameter = function (sParam) {
+        return getUrlParameterFromString(sParam, window.location.search);
+    };
+    var getUrlParameterFromString = function (sParam, sSearch) {
+        var sPageURL = decodeURIComponent(sSearch.substring(1)),
+            sURLVariables = sPageURL.split('&'),
+            sParameterName,
+            i;
+
+        for (i = 0; i < sURLVariables.length; i++) {
+            sParameterName = sURLVariables[i].split('=');
+
+            if (sParameterName[0] === sParam) {
+                return sParameterName[1] === undefined ? true : sParameterName[1];
```

Comment:
> Did you see above that it says `decodeURIComponent` so `sPageURL` is decoded?

Interesting. How does that work if there's an `=` or `&` in the parameter value? It seems the way you're splitting the string here would not work properly if the values are not URL-encoded.